clippy_lints/src/unwrap.rs

```diff
@@ -292,6 +292,10 @@ impl<'tcx> Visitor<'tcx> for UnwrappableVariablesVisitor<'_, 'tcx> {
         if expr.span.in_external_macro(self.cx.tcx.sess.source_map()) {
             return;
         }
+        // Skip checking all expressions inside closures
```

Comment:
Please include the reason for doing so in the comment:

```suggestion
        // Skip checking inside closures since they are visited through `Unwrap::check_fn()` already.
```